applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.hpp

```
@@ -0,0 +1,187 @@
+
+        // Check for UMAT PHI Parameter
+        KRATOS_ERROR_IF(!part_properties.Has(INDEX_OF_UMAT_PHI_PARAMETER) ||
+                        !part_properties.Has(NUMBER_OF_UMAT_PARAMETERS) || !part_properties.Has(UMAT_PARAMETERS))
+            << "Insufficient material data for C-Phi reduction process: " << std::endl;
+        KRATOS_ERROR_IF(part_properties[INDEX_OF_UMAT_PHI_PARAMETER] < 1 || part_properties[INDEX_OF_UMAT_PHI_PARAMETER] > part_properties[NUMBER_OF_UMAT_PARAMETERS])
+            << "undefined INDEX_OF_UMAT_PHI_PARAMETER: " << part_properties[INDEX_OF_UMAT_PHI_PARAMETER]
+            << std::endl;
+        const double phi =
+            part_properties[UMAT_PARAMETERS][part_properties[INDEX_OF_UMAT_PHI_PARAMETER] - 1];
+        KRATOS_ERROR_IF(phi < 0. || phi > 90.) << "Friction angle Phi out of range: " << phi << std::endl;
+        return phi;
+    }
+
+    double ComputeReducedPhi(double Phi) const
+    {
+        // Phi converted to radians and then its tangent is reduced by the reduction factor
+        const double tan_phi         = std::tan(MathUtils<>::DegreesToRadians(Phi));
+        const double reduced_tan_phi = tan_phi * mReductionFactor;
+        return std::atan(reduced_tan_phi) * 180.0 / Globals::Pi;
+    }
+
+    double GetAndCheckC(const Element::PropertiesType& rProp)
+    {
+        // Get the initial properties from the model part. Recall that we create a separate
+        // properties object with reduced c and phi for each and every element. Those reduced
+        // properties objects are not linked to the original ones.
+        const auto& part_properties = mrModelPart.GetProperties(rProp.Id());
+        KRATOS_ERROR_IF(!part_properties.Has(INDEX_OF_UMAT_C_PARAMETER) ||
+                        !part_properties.Has(NUMBER_OF_UMAT_PARAMETERS) || !part_properties.Has(UMAT_PARAMETERS))
+            << "Insufficient material data for C-phi reduction: " << std::endl;
```

As suggested above, I would split these checks, just to raise a better error message if some data is missing.

---

expanded into three lines